p8e-api/src/main/kotlin/io/provenance/engine/stream/ScopeStream.kt

```diff
@@ -106,18 +87,15 @@ class ScopeStream(
         val record = transaction { EventStreamRecord.findById(eventStreamId) }
         val lastHeight = record?.lastBlockHeight
             ?: transaction { EventStreamRecord.insert(eventStreamId, epochHeight) }.lastBlockHeight
-        val request = EventProtos.EventStreamReq.newBuilder()
-            .addAllEventTypes(eventTypes)
-            .setStartHeight(lastHeight + 1)
-            .setConsumer(StreamClientParams(uri, apiKey).consumer)
-            .build()
-        val responseObserver = EventStreamResponseObserver<EventProtos.EventBatch>() { batch ->
+
+        val responseObserver = EventStreamResponseObserver<EventBatch>() { batch ->
```

Comment:
That's cool that you transformed the json format into what we were already using here and it's even using the StreamObserver.

Reply:
yeah, seemed best to avoid as many changes here as possible at least for now... though I don't know if it is weird that the EventStreamResponseObserver still implements the grpc StreamObserver interface, given that this isn't interacting directly with grpc... I might remove that...